docs/specs/tsd-002-custom-builder.md

```diff
@@ -0,0 +1,201 @@
+# Telegraf Custom-Builder
+
+## Objective
+
+Provide a tool to build a customized, smaller version of Telegraf with only
+the required plugins included.
+
+## Keywords
+
+tool, binary size, customization
+
+## Overview
+
+The Telegraf binary continues to grow as new plugins and features are added
+and dependencies are updated. Users running on resource constraint systems
+such as embedded-systems or inside containers might suffer from that grow.
+
+This document specifies a tool to build a smaller Telegraf binary tailored to
+the plugins configured and actually used, removing unnecessary and unused
+plugins. The implementation should be able to cope with configured parsers and
+serializers including defaults for those plugin categories.
+
+Please note, the customization tool might not be available for older versions
+of Telegraf. Furthermore, the degree of customization and thus the effective
+size reduction might vary across versions.
+
+The specified tool will *not* produce distribution packages or containers but
+*only* the customized, static Telegraf binary.
+
+Requirements to produce a customized Telegraf binary are listed below.
+
+### Configuration files
+
+The user has to provide one or more valid Telegraf configuration files or
+configuration directories that are used with the produced binary later on. If
+you plan to use the customized binary in multiple scenarios, the configuration
+files should be a superset of all expected use-cases.
+
+### Telegraf source code
+
+To build a customized version of Telegraf you need access to the [Telegraf
+source-code repository](https://github.com/influxdata/telegraf).
+
+### Build tools
+
+For compiling the customized binary you need the [Golang language](https://go.dev/)
+as well as the `make` build system. The minimum required version of Golang
+can be found in the *Build From Source* section of the `README.md` file of your
+version. Both the `go` and the `make` command must be available in your path.
+
+## Workflow
+
+The first step is to download the Telegraf repository for the version you are
+planning to customize.
+
+```shell
+$ git clone  --branch v1.29.5 --single-branch https://github.com/influxdata/telegraf.git
```

Comment:
```suggestion
$ git clone --branch v1.29.5 --single-branch https://github.com/influxdata/telegraf.git
```

Reply:
I'm a little concerned someone will see 1.29.5 and think that is the only version that supported this. Maybe we show building from master and mention you can check out a specific release branch?

Reply:
TBH I think users that take the effort to customize their builds are usually more likely to be in an enterprise environment and will most likely use some released version... As a compromise, how about explicitly showing both, maybe even with `v<version of telegraf>` instead of a concrete version number?